doc/md/ddl.md

```diff
@@ -116,155 +116,122 @@ playlist "comedy" {
-  "ariga.io/atlas/schema/schemaspec/schemahcl"
-)
-
-func TestSeinfeld(t *testing.T) {
-	f := `
-  show "seinfeld" {
-      writers = [
-          "Jerry Seinfeld",
-          "Larry David",
-      ]
-  }`
-
-	s, err := schemahcl.Decode([]byte(f))
-	require.NoError(t, err)
-	seinfeld := s.Children[0]
-	require.EqualValues(t, &schemaspec.Resource{
-		Type: "show",
-		Name: "seinfeld",
-		Attrs: []*schemaspec.Attr{
-			{
-				K: "writers",
-				V: &schemaspec.ListValue{
-					V: []schemaspec.Value{
-						&schemaspec.LiteralValue{V: `"Jerry Seinfeld"`},
-						&schemaspec.LiteralValue{V: `"Larry David"`},
-					},
-				},
-			},
-		},
-	}, seinfeld)
+func ExampleUnmarshal() {
+    f := `
+show "seinfeld" {
+	writer "jerry" {
+		full_name = "Jerry Seinfeld"	
+	}
+	writer "larry" {
+		full_name = "Larry David"	
+	}
+}`
+
+  type (
+      Writer struct {
+        ID       string `spec:",name"`
+        FullName string `spec:"full_name"`
+      }
+      Show struct {
+        Name    string    `spec:",name"`
+        Writers []*Writer `spec:"writer"`
+      }
+    )
+  var test struct {
+    Shows []*Show `spec:"show"`
+  }
+  err := Unmarshal([]byte(f), &test)
+  if err != nil {
+    panic(err)
+  }
+  seinfeld := test.Shows[0]
+  fmt.Printf("the show %q has %d writers.", seinfeld.Name, len(seinfeld.Writers))
+  // Output: the show "seinfeld" has 2 writers.
 }
+
 ```
 
-Observe that `Decode` returns a `schemaspec.Resource`. This Go type is
-a generic container for resources described in the Atlas DDL. Applications
-are not expected to work with it directly as it is not structured and 
-not type-safe. In the section about [Extensions](#Extensions), we discuss the way applications can
-work with data coming from Atlas HCL documents. 
+Observe that similar to the standard-library's `json.Unmarshal` function, this function
+takes as arguments a byte-slice and an empty interface.  The empty interface should be a
+pointer to a struct into which the `Unmarshal` function will read the values. The struct
+fields must be annotated with `spec` tags that define the mapping from HCL to the Go type.
+This mapping is discussed in the section about [Extensions](#Extensions).
 
 ### Writing with Go
 
-To encode `schemaspec.Resource` instances back into HCL, use the `schemahcl.Encode` 
-([doc](https://pkg.go.dev/ariga.io/atlas@v0.0.0-20211004124157-99ae6aaad16b/schema/schemaspec/schemahcl#Encode)) function:
+To encode a Go struct back into HCL, use the `schemahcl.Marshal`
+([doc](https://pkg.go.dev/ariga.io/atlas/schema/schemaspec/schemahcl#Marshal)) function:
 
 ```go
-func TestSeinfeldEncode(t *testing.T) {
-	r := &schemaspec.Resource{
-		Children: []*schemaspec.Resource{
-			{
-				Name: "seinfeld",
-				Type: "show",
-			},
-		},
-	}
-	encode, err := Encode(r)
-	require.NoError(t, err)
-	expected := `show "seinfeld" {
-}
-`
-	require.EqualValues(t, expected, string(encode))
+func ExampleMarshal() {
+  type (
+    Point struct {
+      ID string `spec:",name"`
```

Comment:
I prefer the `atlas` word over `spec`, how can we check that it's not used anywhere else?

Reply:
Let's discuss this and do it in a separate PR.